### ISSUE TITLE
Quick Start Guide for New Contributors Guide

### ISSUE THREAD

**Issue by AntonLandor** [dotnet/corefx#41729]
The changes in #41555 are great since the wiki was really outdated and lacking. However, as I quite recently was a new contributor I would like to see some short and concise _Quick Start Guide_ or similar since that would be really helpful for me when I was new here. When first looking into this project it may seem very overwhelming to make your first contribution, and a simple quick start guide could be good to get the new contributors on track faster. 

Of course, they need to read some of the more elaborate documents to make contributions, but the _Quick Start Guide_ could serve as a simple introduction to get the contributors up and running and familiar with the most basic parts of developing for this project. I could try to make a first draft for this _Quick Start Guide_ if it's deemed appropriate to add to this project.

**Comment by danmosemsft** [comment]
@carlossanlop

**Comment by danmosemsft** [comment]
@AntonLandor what would such a quick start look like? Would it cover all OS? Would it duplicate existing docs or just link to them? Would it walk through an example change?

Always interested in making things easier, just not clear how this would be not just yet another doc.?

**Comment by AntonLandor** [comment]
You're right, we want to avoid redundant documentation since that will be just another doc to maintain. I haven't completely thought out the structure, but will of course if this doc should be written. 

I think I have an alright perspective on how new developers approach this project since I recently started contributing. When I first started developing I liked how the wiki provided an outline for what was required to make new contributions, clearly stating the steps including, for example, _Prerequisites_, _Building the project_, _Testing_, etc. I think the quick start guide could just contain a suggested step by step process for getting started with one's first contribution, and I think it's important to keep this short and concise to make it clear for new devs what each step is. Every step in itself could link to the more elaborate documents. Steps could be for example: 
1. Install prerequisites ([Windows](https://github.com/dotnet/corefx/blob/master/Documentation/building/windows-instructions.md#required-software), [Linux](https://github.com/dotnet/corefx/blob/master/Documentation/building/unix-instructions.md#user-content-linux), [macOS](https://github.com/dotnet/corefx/blob/master/Documentation/building/unix-instructions.md#user-content-macos))
2. Fork and clone repository (_Linking to another doc with more elaborate instructions_)
3. Run the build script in the root folder of CoreFX (`build.cmd` on windows, `./build.sh` on Unix)
4. [Build and run tests](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md#build-and-run-tests)
... and so on.

> Would it cover all OS?

Ideally, yes. Though I'm not familiar with developing on macOS, but I think I'll be able to extract the required information from other docs.

> Would it duplicate existing docs or just link to them?

Mostly link to them, but if there is some information that could be presented easier in this _Quick Start Guide_ (has to be short then) I'll use this information directly in the guide.

> Would it walk through an example change?

Yes, that seems like a good idea to help the developer to exemplify how changes are generally made. 

> Always interested in making things easier, just not clear how this would be not just yet another doc.?

Hopefully, it will help new developers sort out what they should do first to get up an running, without getting confused by several long documents.

I could ask some other students who are not familiar with this project to give their input.

**Comment by ViktorHofer** [comment]
For that we have the `Contributing.md` file which should either have much more information or point to another file. See https://mozillascience.github.io/working-open-workshop/contributing/ for more details.

I think merging a simplified version of developer-guide.md with the CONTRIBUTING.md would make most sense here. The additional details from the developer-guide should be placed into sub-files into the corresponding categories (workflow categories).

**Comment by AntonLandor** [comment]
> For that we have the `Contributing.md` file which should either have much more information or point to another file. See https://mozillascience.github.io/working-open-workshop/contributing/ for more details.
> 
> I think merging a simplified version of developer-guide.md with the CONTRIBUTING.md would make most sense here. The additional details from the developer-guide should be placed into sub-files into the corresponding categories (workflow categories).

@ViktorHofer Yes, I agree it should be in `Contributing.md`. You could add some of the parts discussed in the [page you linked](https://mozillascience.github.io/working-open-workshop/contributing/). What I think is lacking right now for new developers is some simple outline for how the workflow could look, and what they should do first to get started. There are great docs which can be linked in `Contributing.md`, so this will most likely not add too much redundant information that has to be maintained.
